### ISSUE TITLE
Linking error with intel compilers

### ISSUE THREAD

**Issue by vvedula22** [SimVascular/svFSI#43]
I am trying to compile svFSI using Intel compilers on Expanse but it fails during linking step:

[100%] Linking CXX executable ../../bin/svFSI
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crt1.o: In function `_start':
(.text+0x24): undefined reference to `main'
make[5]: *** [Source/svFSI/CMakeFiles/svFSI.dir/build.make:1569: bin/svFSI] Error 1
make[4]: *** [CMakeFiles/Makefile2:453: Source/svFSI/CMakeFiles/svFSI.dir/all] Error 2
make[3]: *** [Makefile:149: all] Error 2
make[2]: *** [CMakeFiles/svFSI.dir/build.make:132: svFSI-prefix/src/svFSI-stamp/svFSI-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:124: CMakeFiles/svFSI.dir/all] Error 2
make: *** [Makefile:103: all] Error 2

Modules loaded are shown below:

```
$ module purge
$ module load DefaultModules cpu intel intel-mpi intel-mkl cmake
$ module list

Currently Loaded Modules:
  1) shared                  3) DefaultModules   5) intel/19.1.1.217       7) intel-mkl/2019.1.144
  2) slurm/expanse/20.02.3   4) cpu/0.15.4       6) intel-mpi/2019.8.254   8) cmake/3.18.2
```

**Comment by ktbolt** [comment]
@vvedula22 I've not seen this error before. This builds ok using using gcc and openmpi? Be sure to remove your build directory when rebuilding.

**Comment by vvedula22** [comment]
@ktbolt GNU compilers build fine either with openmpi or mvapich2. I get this error when I use Intel compilers. Would like to know if anyone who used Intel compilers faced a similar error.

**Comment by CZHU20** [comment]
Hi @vvedula22, I submitted a pull request a couple weeks ago, in which the build system is modified to be compatible with Intel oneAPI compiler suite.  I also had some linking issues, and it turned out that we needed to explicitly tell the linker that we are linking objects built from C++ code and Fortran code.

**Comment by vvedula22** [comment]
@CZHU20 I didn't realize these were related. Will check if your pull request helps me compile with intel compilers on Expanse.

**Comment by vvedula22** [comment]
The issue has to do with linking BLAS and LAPACK libraries in addition to MKL. This is fixed in pull request #56 for Intel oneAPI support.
